packages/terra-show-hide/src/ShowHide.jsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Toggle from 'terra-toggle';
+import { injectIntl, intlShape } from 'terra-base';
+import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './ShowHide.module.scss';
+import Button from './_ShowHideButton';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Content in the body of the component that will be shown or hidden
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Text that will be visible to the user while the component is collapsed
+   */
+  preview: PropTypes.string.isRequired,
+  /**
+   * Button text that will be displayed when the component is collapsed
+   */
+  collapsedButtonText: PropTypes.string,
+  /**
+   * Button text that will be displayed when the component is expanded
+   */
+  expandedButtonText: PropTypes.string,
+  /**
+   * Icon displayed next to the button text
+   */
+  icon: PropTypes.element,
+  /**
+   * The intl object to be injected for translations. Provided by the injectIntl function.
+   */
+  intl: intlShape.isRequired,
+  /**
+   * Sets the animation for the component when it is expanded or collapsed
+   */
+  isAnimated: PropTypes.bool,
+  /**
+   * Sets the initial state of the component to open and expanded
+   */
+  isInitiallyOpen: PropTypes.bool,
+  /**
+   * Callback function triggered when the component is collapsed
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function triggered when the component is expanded
+   */
+  onOpen: PropTypes.func,
+};
+
+const defaultProps = {
+  icon: <IconChevronDown />,
+  isAnimated: false,
+  isInitiallyOpen: false,
+};
+
+class ShowHide extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: props.isInitiallyOpen,
```

Comment:
Should we be doing this as an uncontrolled component? I've seen apps in Cerner that would have a button that would open and expand all of the menus and panels at the same time. If we don't want that, we should at least have a way to change this state value from outside the component. https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#new-lifecycle-getderivedstatefromprops

Reply:
+1 to making this component uncontrolled. I think if you do so, you could also go from having two callbacks (onOpen on onClose) to a single callback (onChange/onToggle/onSomething?). The component could decide what needs to happen in that callback based on the isOpen state (which they would own).

Reply:
I think I confused you as well. Sorry about that.

Right now, the component is uncontrolled as the isOpen status is maintained inside this component. For a controlled component, the parent would store the state of isOpen, and this state I believe wouldn't even be set. The request is to go with the controlled route, where instead of this.state.isOpen, you would look at this.props.isOpen

Reply:
If you would like an example of the different components, check out our paginator code. https://github.com/cerner/terra-core/tree/master/packages/terra-paginator

Reply:
No worries about the terminology mixup, I do it too. I am still confused about this thread though. @tb5748 said in his comment that after this refactor, the component should not have internal state. But the terra-paginator example that @JakeLaCombe provided manages its own state variables

Reply:
The ControlledPaginator is what I think Jake was referring to: https://github.com/cerner/terra-core/blob/master/packages/terra-paginator/src/ControlledPaginator.jsx

Reply:
@tbiethman thanks for the clarification!

Reply:
@JakeLaCombe @tbiethman  the refactor is ready for review: https://github.com/cerner/terra-core/pull/1910/commits/6979d941a0a0749df9f9ffeae82def7c0b9f34c5